### PR TITLE
[Janus-Pro] Add e2e text-to-image pipeline (1B + 7B): nightly + benchmark

### PR DESCRIPTION
**Pull request by ashokkumarkannan1**
### Ticket

- https://github.com/tenstorrent/tt-xla/issues/5233

### Problem description

- Janus-Pro passed component bring-up — `image_token_step`, `gen_img_embed` and `gen_vision_decode` for both the 1B and 7B variants (added in #4810) — but there was no end-to-end text-to-image pipeline wired into nightly + benchmark CI. 
- This mirrors the image-gen pattern introduced in #5044 (Playground v2.5) and extended by SDXL-Lightning: a model whose components pass bring-up should get an e2e pipeline in nightly + a perf benchmark.

### What's changed

Chains the three validated Janus-Pro components into an end-to-end text-to-image pipeline on TT, following the deepseek-ai/Janus `generation_inference.py` reference loop (classifier-free guidance + multinomial sampling, 576 image tokens, `parallel_size=1`) and the `playground_v2_5` / `sdxl_lightning` e2e-pipeline conventions.

- **Nightly** (`tests/torch/models/janus_pro/test_janus_pro_pipeline.py`): self-contained — pipeline class + `run_janus_pro_pipeline` helper + tests inline (no import from `examples/`). `test_janus_pro_pipeline` (1B) and `test_janus_pro_pipeline_7b` (7B). Markers: `nightly`, `model_test`, `single_device`, `large`, `record_test_properties`. The autoregressive step uses a `StaticCache` pre-allocated to `prompt_len + 576` so the prefill and decode graphs compile **once** and every later decode step is a cache hit (a `DynamicCache`, as in the original script, recompiles each step because the KV length changes — hours on TT). Components move to TT via `model.compile(backend="tt") + model.to(xla_device())` and are evicted back to CPU after use, keeping at most one model resident on TT DRAM.
- **Benchmark**: a dedicated AR harness `tests/benchmark/benchmarks/ar_imagegen_benchmark.py` (sibling of the diffusion `imagegen_benchmark.py` — the diffusion harness is hard-wired to `te1/te2/unet_steps/vae`, so AR needs its own `_perf` schema and reports decode **tokens/second**) + a self-contained `benchmarks/janus_pro_pipeline.py` with per-stage `_perf` timing. Entries `test_janus_pro` (1B) and `test_janus_pro_7b` (7B) in `test_imagegen.py`. Two-pass warmup + steady-state. `optimization_level=0` (opt_level=1 stalls the Janus LM/vision compile).
- **Matrix**: entries `janus_pro_1b` and `janus_pro_7b` added to `.github/workflows/perf-bench-matrix.json`.

Both variants validated on blackhole (p150). **7B requires blackhole** — it OOMs the DRAM on wormhole (n150), same as the Pro-7B `ImageTokenStep` component test.

| Test | Result | Time |
| --- | --- | --- |
| `test_janus_pro_pipeline` (1B, nightly) | ✅ passed | ~2m15s |
| `test_janus_pro_pipeline_7b` (7B, nightly) | ✅ passed | ~6m18s |
| `test_janus_pro` (1B, benchmark) | ✅ passed | ~3m29s |
| `test_janus_pro_7b` (7B, benchmark) | ✅ passed | ~6m48s |

Steady-state benchmark numbers (blackhole, opt_level=0):

| | 1B | 7B |
| --- | --- | --- |
| e2e latency | 86.0s | 161.6s |
| decode tokens/s | 11.36 | 6.82 |
| prefill | 25.2s | 50.0s |
| vision decode | 7.9s | 8.1s |

### Checklist

- [x] Validate benchmark test on p150 (blackhole) — https://github.com/tenstorrent/tt-xla/actions/runs/27828742737
- [x] Attached passing logs + output images here for reference - [janus_pro_artifacts.zip](https://github.com/user-attachments/files/29138729/janus_pro_artifacts.zip)

### CODE REVIEW

**Review by nsumrakTT — APPROVED** [review]

File: .github/workflows/perf-bench-matrix.json [hunk 1]
```diff
@@ -70,6 +70,14 @@
         "name": "sdxl_lightning",
         "pytest": "tests/benchmark/test_imagegen.py::test_sdxl_lightning"
       },
+      {
+        "name": "janus_pro_1b",
+        "pytest": "tests/benchmark/test_imagegen.py::test_janus_pro"
```
Comment (kamalrajkannan78) [inline .github/workflows/perf-bench-matrix.json:75]: test in n150 too. if it fails change (set "runs-on": "p150" for the model entry in `.github/workflows/perf-bench-matrix.json`) - [reference](https://github.com/tenstorrent/tt-xla/pull/5244#pullrequestreview-4516333525)

**Review by kamalrajkannan78 — APPROVED** [review]

File: .github/workflows/perf-bench-matrix.json (hunk 1 above)
Comment (ashokkumarkannan1) [reply .github/workflows/perf-bench-matrix.json:75]: Confirmed — Janus-Pro-7B OOMs the DRAM on wormhole (n150); it only fits on blackhole (p150).

- Verified on CI by running the same benchmark on n150: `janus_pro_1b` passes and `janus_pro_7b` fails with OOM — https://github.com/tenstorrent/tt-xla/actions/runs/27933115238

Added a wormhole skip for 7B in this PR
- The 7B nightly pipeline test (`test_janus_pro_pipeline_7b`) and the 7B benchmark (`test_janus_pro_7b`) now skip on n150 (wormhole), reusing the existing `skip_pro_7b_image_token_on_wormhole` guard pattern.
- The `janus_pro_7b` matrix entry is pinned to `p150-perf` so CI never schedules it on n150.

1B continues to run on both n150 and p150.

**Review by vmilosevic — APPROVED** [review]

**Review by vvukomanTT — APPROVED** [review]

**Review by sdjukicTT — APPROVED** [review]

**Review by vkovacevicTT — APPROVED** [review]
